src/com/com_nrf24l01_spi_protocol.cpp

```diff
@@ -0,0 +1,106 @@
+#include "com_nrf24l01_spi_protocol.hpp"
+#include "com_nrf24l01_reg.hpp"
+
+namespace com {
+
+auto NRF24L01SpiProtocol::ReadRegister(const std::uint8_t register_address) noexcept -> std::uint8_t {
+  auto position_in_buffer = 1;
```

Comment:
const is not included in auto. You could also make this constexpr auto and let the compile handle whether it is true constexpr or only const

Reply:
Done.